src/com/intel/checkoutScmInternal.groovy

```diff
@@ -0,0 +1,74 @@
+// src/com/intel/checkoutScmInternal.groovy
+
+package com.intel
+
+/**
+ * checkoutScmInternal.groovy
+ *
+ * Simplified Scm checkout routine.
+ */
+
+
+def checkoutScmInternal(Map config = [:]) {
+  /**
+   * Simplified SCM checkout method.
+   *
+   * @param config Map of parameters passed
+   * @return Map of any variables the SCM plugin would set in a
+   * Freestyle job.
+   *
+   * config['scm'] SCM name, defaults to 'GitSCM'.
+   * config['url'] Url for SCM repository.  If not specified,
+   *  defaults will be used from the scm global variable.
+   * config['cleanAfterCheckout'] True for clean after checkout.
+   * config['checkoutDir'] Optional directory to checkout into.
+   * config['branch'] Optional branch to checkout, defaults to master.
+   * config['withSubmodules'] Optional checkout submodules if true.
+   * config['credentialsId'] Optional credentials ID.
+   */
+
+  scm_name = 'GitSCM'
+  if (config['scm']) {
+    scm_name = config['scm']
+  }
+  branches = scm.branches
+  if (config['url']) {
+    userRemoteConfigs = [[url: config['url']]]
+    if (config['branch']) {
+      branches = [[name: config['branch']]]
+    } else {
+      branches = [[name: '*/master']]
+    }
+    if (config['credentialId']) {
```

Comment:
```suggestion
    if (config['credentialsId']) {
```

Reply:
Even with this change, and all of the above ones, I am still getting:

```[checkpatch] userRemoteConfigs: [[url:ssh://review.hpdd.intel.com:29418/exascale/jenkins], [credentialsId:bf21c68b-9107-4a38-8077-e929e644996a]]
[Pipeline] [checkpatch] checkout
Post stage
[checkpatch]  > git rev-parse --is-inside-work-tree # timeout=10
[checkpatch] Fetching changes from the remote Git repository
[checkpatch]  > git config remote.origin.url ssh://review.hpdd.intel.com:29418/exascale/jenkins # timeout=10
[checkpatch] Fetching upstream changes from ssh://review.hpdd.intel.com:29418/exascale/jenkins
[checkpatch]  > git --version # timeout=10
[checkpatch]  > git fetch --tags --progress ssh://review.hpdd.intel.com:29418/exascale/jenkins +refs/heads/*:refs/remotes/origin/*
[checkpatch] ERROR: Error fetching remote repo 'origin'
[checkpatch] hudson.plugins.git.GitException: Failed to fetch from ssh://review.hpdd.intel.com:29418/exascale/jenkins
[checkpatch] 	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:888)
[checkpatch] 	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1155)
[checkpatch] 	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1186)
[checkpatch] 	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
[checkpatch] 	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)
[checkpatch] 	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)
[checkpatch] 	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
[checkpatch] 	at hudson.security.ACL.impersonate(ACL.java:290)
[checkpatch] 	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
[checkpatch] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[checkpatch] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[checkpatch] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[checkpatch] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[checkpatch] 	at java.lang.Thread.run(Thread.java:748)
[checkpatch] Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress ssh://review.hpdd.intel.com:29418/exascale/jenkins +refs/heads/*:refs/remotes/origin/*" returned status code 128:
[checkpatch] stdout: 
[checkpatch] stderr: Permission denied (publickey).
[checkpatch] fatal: Could not read from remote repository.
[checkpatch] 
[checkpatch] Please make sure you have the correct access rights
[checkpatch] and the repository exists.
```

If this patch is working in [your testing of it](https://github.com/daos-stack/scons_local/pull/1) it would be useful to get a fresh test run with the most up-to-date revision of this PR to demonstrate that it is working somewhere.

Reply:
Fixed.

Reply:
fixed.

Reply:
Test runs with this pull review are working for me.